### PR TITLE
Add NFCScanOptions signal tests

### PR DESCRIPTION
**Pull request by xiuqijix**

### CODE REVIEW

File: web-nfc/NFCReader.https.html [hunk 1]
```diff
@@ -131,4 +131,47 @@
   await promise;
 }, "Test that NFCReader.scan succeeds if NFCScanOptions.url is empty.");
 
+nfc_test(async (t, mockNFC) => {
```
Comment (Honry) [inline web-nfc/NFCReader.https.html]: You can also test an invalid AbortSignal.

File: web-nfc/NFCReader.https.html [hunk 2]
```diff
@@ -131,4 +131,47 @@
...
+  controller.abort();
+
+  await new Promise((resolve, reject) => {
+    readerWatcher.wait_for("reading").then(event => {
+      reject("reading event should not be fired.");
+    });
+    t.step_timeout(resolve, 100);
+  });
+
+}, "Test that NFCReader.onreading should be fired if NFCScanOptions.signal is \
```
Comment (Honry) [inline web-nfc/NFCReader.https.html]: This could be simplified with "Synchronously signaled abort."

**Review by Honry — COMMENTED** [review]
Almost all tests in this file are for NFCReader.scan() testing, so we should rename it to something like "NFCReader_scan.https.html"

File: web-nfc/NFCReader.https.html (hunk 2 above)
Comment (kaixinjxq) [reply web-nfc/NFCReader.https.html]: Done.

File: web-nfc/NFCReader.https.html (hunk 1 above)
Comment (kaixinjxq) [reply web-nfc/NFCReader.https.html]: Done.

File: web-nfc/NFCReader.https.html [hunk 3]
```diff
@@ -131,4 +131,47 @@
...
+
+  await new Promise((resolve, reject) => {
+    readerWatcher.wait_for("reading").then(event => {
+      reject("reading event should not be fired.");
+    });
+    t.step_timeout(resolve, 100);
+  });
+
+}, "Test that NFCReader.onreading should not be fired if NFCScanOptions.signal \
+is abort.");
```
[the remark this answers is not in the document]
Comment (kaixinjxq) [reply web-nfc/NFCReader.https.html]: Done.

**Review by Honry — APPROVED** [review]
thanks, LGTM

**Comment by Honry** [comment]
BTW, please rename the file as "NFCReader_scan.https.html".

**Pull request merged**